### DISCUSSION TITLE
How to build a project as an editor

### DISCUSSION THREAD

**Discussion by M-S-M-A-S** [o3de/o3de#13869] [Q&A] [answered]
I'd like to merge a project to the engine so that building the project in turn build the editor, how can i configure cmake to do that??
in another word i want them to be in one solution

**Comment by AMZN-daimini** [comment]
@o3de/sig-build

**Comment by M-S-M-A-S** [comment]
I switched to the development branch and creating new project include the editor source, in the main branch it only included the editor project but the editor.header and editorcore and others weren't included in the solution, i had to attach process to debug which made it difficult to debug

**Comment by lemonade-dm** [comment, marked as answer]
Using a project-centric workflow, allows the Editor to be built in the same solution as the `${project}.GameLauncher` and ${project}.ServerLauncher` applications.

The project-centric workflow involves configuring cmake using the project root directory as the source directory (https://www.o3de.org/docs/user-guide/build/configure-and-build/#configure-with-the-cmake-cli)

For example if your project is in a directory called `D:/NewProject`, then the following command can generator a build solution on Windows
```powershell
PS D:/NewProject> cmake -B build/windows -S.
```
